### DISCUSSION TITLE
Cant Connect to Wrodpress 6.7.1 WebServicesInstallerNotEnabled

### DISCUSSION THREAD

**Discussion by dermonotone** [akeeba/panopticon#866] [Q&A (Community Support)]
Hi there,
i did setup Panopticum and now i tried to add the first site.
After adding the needed Information (URL and Token) i get the response from Panopticum that it cant connect:
Instead, we got an error which we should be able to provide troubleshooting information for (Akeeba\Panopticon\Exception\SiteConnection\WebServicesInstallerNotEnabled), but we don't have the code for it. This should have not happened. 

See screenshot.

Thank you in advance for your help!
Best regards
Harald

<img width="951" alt="pan" src="https://github.com/user-attachments/assets/c0888705-6a28-40ec-a1d5-e62412115969" />

**Comment by nikosdion** [comment]
Your site's WordPress REST API (`/wp-json`) is not accessible. Check your `.htaccess` file, and make sure there is no third party plugin or any modification in your `wp-config.php` file which disables the REST API.

**Comment by dermonotone** [comment]
Hi, Rest API is enabled but on different Endpoint.
its /index.php?rest_route=/ (that is was the panopticum connecttor shows).
When i click that link i see the JSON in Browser.

On a Second Site i get another Error: You did not specify a secret key.
But i did Setup Everything there.

If you need anything to take a closer look let me know.

Thanks again for your help!
Best regards
harald

**Reply by nikosdion** [reply]
The API must be on /wp-json for Panopticon to be able to connect to your site. This requires having URL rewriting enabled. If yo are using NginX or IIS you can still do that, but it requires manual configuration which is beyond what I will help you with.

**Comment by dermonotone** [comment]
Hi,
thank you okay. i will check that.

But i tried now to add another 2 sites and getting this error: You did not specify a secret key.
I entered all information from the connector to panopticum. Is there something i do wronmg?
Rest api is working for both sites.

thanks again
Harald
<img width="1002" alt="pan2" src="https://github.com/user-attachments/assets/a24291fe-40af-42fb-8d0b-e97f14f7ef55" />

**Reply by nikosdion** [reply]
Are you sure you're using version 1.3.1?

**Comment by dermonotone** [comment]
hi,
yes i run 1.3.1.
i also have problems with the one site i was able to connect. i installed admin tools and backup but it does not work.
any help?
thanks ain advance
<img width="968" alt="pn_error" src="https://github.com/user-attachments/assets/e25c869e-f732-4686-bf08-8fb80167085b" />

**Reply by nikosdion** [reply]
Do you have the Pro versions of both Akeeba Backup and Admin Tools?

**Comment by dermonotone** [comment]
yes, i have pro of both :-)

**Reply by nikosdion** [reply]
I can help you troubleshoot your issue, but I need the following connection information to your site. I will take a backup of your site, restore on my dev server, and try to reproduce your issue in a controlled environment.

**Site connection information**

* Login URL e.g. `https://www.example.com/administrator/index.php?secret123`
* Super User username
* Super User password

**(S)FTP connection information**

* Hostname
* Port (if other than 21 or 22)
* Username
* Password

Do **NOT** paste this information unencrypted. GitHub issues and discussions are publicly visible. Instead, follow the instructions below to send me an encrypted copy of the information which can only be decrypted by me:

* Go to https://keybase.io/encrypt
* In the **Recipient** field type `nikosdion`.
* It will show you my face, my name, and the fact that I have linked my same-named GitHub account; verify this information and click on my face.
* In the **Message to encrypt** area enter the information I asked you above.
* Click the Encrypt button.
* After a short while the text area changes its title to **The secret message**.
* Click in the text area to copy the secret (encrypted) message.
* Paste it to your next reply to me.

This is perfectly safe to do! It uses public key encryption, just like HTTPS, with a _much, MUCH stronger cipher_ (3744-bit Elliptic Curve Cryptography which is insanely secure). The message can only be decrypted with my secret key which is safely stored on a hardware token, protected by a PIN.

Please make sure that you test the access credentials using a separate browser / (S)FTP software than what you normally use to ensure that they really work. Please make sure that there are no geographic or other restrictions which may prevent me from accessing your site. Kindly note that I am on the [Asia/Nicosia timezone](https://www.timeanddate.com/worldclock/cyprus/nicosia) and I am handling support free of charge; some delays are to be expected.

**Comment by dermonotone** [comment]
hi,
here is the secret message:

-----BEGIN PGP MESSAGE-----
Version: Keybase OpenPGP v2.1.13
Comment: https://keybase.io/crypto

wcBMA6+1LLjxrhABAQf+OcluadpMuop7PYefn72+FK4yL7d3tw1W2xIhJh+q0jjX
DGTIh9VlLTuWG/xbUaWwHwRrs5gyM/JvE7RP2TYajh8XbpEJ6LcXuMX1pDhNIh1q
+KR4uWUpfFOcWy1QskkF2Fn2rBOca/to8u9Tjs6rlrDaU4phjDNmToQninQM11GJ
Ihi91nfoCuS88bdCx+VvZSvp7m7HpZFWp/2nM5X1ZT1p2NdrECG+OmxfJYCRUukf
l0Rx2CCgLQSIIkRmtMJCIv8uO+1Eg3ztxlp8cIIOk18jVyYTxuf38Ra3DvIlcLMZ
/vBjzju3Sv5ukK+GjmibFsRV1Y2N5oCNFveaGmdu0NLBUAHnHcALGSUwYs0C0ySL
NqkIGg29hXS4M5jnxj5ckECMBybQtHHkj8sgLphCX9qpmQA52fFvEhlF74sw8BXQ
eNmCQ0C3JIcKfQf3YyUNiAOey87N+t6Wp0Htlsrb2+3nFwc0W/9kNaPJ0NGldB4Q
ML9wGnk4I4/5pFYcJOgbW3/jUYl+og+RQiSaRVhP1jvnCepRLTxInaUegXam8R35
mVnXyLEQXISvtELJNHntSmbt+0UAL4zg/Vz5MTGXlVPZFqCzecFqxVkla2fG4yD3
BZTCP0bvsQHDcfOlRGXLzFfPMFAnkbV5E8qOdBJ6nXhDK6v0iRTAI9lckq77O6kz
h8cFvYLQ1xcux9hGkVp1jCMEONGT20d0zZi6Qve9oitkCXjEjG4jyF7LmJ2zovUW
tRSPdxiX804+OEprMw/pOck56ICgkLP6cRJEgpZVrIncTcGg1isbbJBhLIE1ECmT
lfj0fc1p1u6HE5OuEMKaUXe71CptpCoXwvpj+E60zpGOCkpacAJSJYyX8EOA01Z1
o/UG1dICgwRdaJDxKXYe/MVwE/7LvBq1Y/QjSBiGZ6knjx9NuZFZgbgbGgL1wQoB
vaWlgAon3ZftugVieQ4kzdSIMOXBZNWvxx6e0B06rQAawvEjRra9WKJ1iLzW7aFO
PkR+/daYxdyqaFL5k2bLPhge58fOMXs4qrNPCnTFGdcwRA==
=depu
-----END PGP MESSAGE-----


Thanks again for your support! And pls let me know if ypu need anything else.
Best regards

**Reply by nikosdion** [reply]
Both sites: Remote backup was turned off in Akeeba Backup. Go to Akeeba Backup, System Configuration, Public API and set “Enable JSON API (remote backup)” to Yes. Enter a new “Secret key”; you can create a new one securely in https://www.random.org/passwords/?num=1&len=32&format=html&rnd=new  There is a known issue in Panopticon where it will be impossible to connect a WordPress site which has Akeeba Backup Professional installed, but its remote JSON feature is disabled and/or there is no configured secret key for it. This is on my to-do list; I am currently rewriting the backup restoration script, which takes priority.

Second site: You have installed the **wrong file** for the connector plugin. Delete the folder wp-content/plugins/panopticon-connector-wordpress-1.0.2. Go to https://github.com/akeeba/panopticon-connector-wordpress/releases/latest. Scroll all the way down. Expand Assets. Download panopticon-1.0.2.zip. Install it on your site. Activate the plugin. Remember that you will also need to follow the instructions in the previous paragraph.

PS: While restoring the second site I confirmed an issue I had spotted a few days ago while rewriting the restoration script. When restoring the site on my dev server the restoration "hang" because the old site is using a remote database server hostname and port which blackholes the connection instead of returning a proper error that we are not allowed to connect to it. There's a workaround coming for this in the next version of our backup software. It does not affect your connection to Panopticon, I am just mentioning it because it's something I did ran across while restoring the backup of your site locally.
